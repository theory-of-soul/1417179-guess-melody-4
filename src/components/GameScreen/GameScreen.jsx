import React from "react";
import PropTypes from "prop-types";

const GameScreen = ({children}) => {

  return (
    <section className="game game--artist">
      <header className="game__header">
        <a className="game__back" href="#">
          <span className="visually-hidden">Сыграть ещё раз</span>
          <img className="game__logo" src="img/melody-logo-ginger.png" alt="Угадай мелодию" />
        </a>

        {/*  <svg xmlns="http://www.w3.org/2000/svg" className="timer" viewBox="0 0 780 780">*/}
        {/*  <circle className="timer__line" cx="390" cy="390" r="370"*/}
        {/*    style="filter: url(#blur); transform: rotate(-90deg) scaleY(-1); transform-origin: center"/>*/}
        {/*  </svg>*/}

        <div className="game__mistakes">
          <div className="wrong"></div>
          <div className="wrong"></div>
          <div className="wrong"></div>
        </div>
      </header>

      <section className="game__screen">
        {children}
      </section>
    </section>
  );
};

GameScreen.propTypes = {
  children: PropTypes.element.isRequired
};

export default GameScreen;
