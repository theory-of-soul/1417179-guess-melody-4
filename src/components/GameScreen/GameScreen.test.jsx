import React from "react";
import renderer from 'react-test-renderer';
import GameScreen from "./GameScreen";

const children = <div/>;

describe(`GameScreen snapshot tests`, () => {
  it(`GameScreen rendered template with children`, () => {
    const component = renderer
      .create(<GameScreen>{children}</GameScreen>)
      .toJSON();

    expect(component).toMatchSnapshot();
  });
});
