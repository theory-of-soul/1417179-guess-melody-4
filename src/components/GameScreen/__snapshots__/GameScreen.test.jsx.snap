// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GameScreen snapshot tests GameScreen rendered template 1`] = `
<section
  className="game game--artist"
>
  <header
    className="game__header"
  >
    <a
      className="game__back"
      href="#"
    >
      <span
        className="visually-hidden"
      >
        Сыграть ещё раз
      </span>
      <img
        alt="Угадай мелодию"
        className="game__logo"
        src="img/melody-logo-ginger.png"
      />
    </a>
    <div
      className="game__mistakes"
    >
      <div
        className="wrong"
      />
      <div
        className="wrong"
      />
      <div
        className="wrong"
      />
    </div>
  </header>
  <section
    className="game__screen"
  />
</section>
`;

exports[`GameScreen snapshot tests GameScreen rendered template with children 1`] = `
<section
  className="game game--artist"
>
  <header
    className="game__header"
  >
    <a
      className="game__back"
      href="#"
    >
      <span
        className="visually-hidden"
      >
        Сыграть ещё раз
      </span>
      <img
        alt="Угадай мелодию"
        className="game__logo"
        src="img/melody-logo-ginger.png"
      />
    </a>
    <div
      className="game__mistakes"
    >
      <div
        className="wrong"
      />
      <div
        className="wrong"
      />
      <div
        className="wrong"
      />
    </div>
  </header>
  <section
    className="game__screen"
  >
    <div />
  </section>
</section>
`;
